packages/core/src/codewhisperer/util/supplementalContext/utgUtils.ts

```diff
@@ -184,41 +186,45 @@ async function getRelevantUtgFiles(editor: vscode.TextEditor): Promise<string[]>
     })
 }
 
+export function guessSrcFileName(
+    testFileName: string,
+    languageId: vscode.TextDocument['languageId']
+): string | undefined {
+    const languageConfig = utgLanguageConfigs[languageId]
+    if (!languageConfig) {
+        return undefined
+    }
+
+    for (const pattern of languageConfig.testFilenamePattern) {
+        try {
+            const match = testFileName.match(pattern)
+            if (match) {
+                return match[1] + match[2]
+            }
+        } catch (err) {
+            if (err instanceof Error) {
+                getLogger().debug(
```

Comment:
is it expected that an exception would be thrown? if not, it indicates a bug, and that should be logged at "error" level. 

```suggestion
                getLogger().error(
```

Reply:
yea that makes sense, updating it.

Reply:
updated